pilot/pkg/config/kube/ingress/conversion.go

```diff
@@ -216,20 +216,22 @@ func ConvertIngressVirtualService(ingress v1beta1.Ingress, domainSuffix string,
 		if f {
 			vs := old.Spec.(*networking.VirtualService)
 			vs.Http = append(vs.Http, httpRoutes...)
-			sort.SliceStable(vs.Http, func(i, j int) bool {
-				r1 := vs.Http[i].Match[0].GetUri()
-				r2 := vs.Http[j].Match[0].GetUri()
-				_, r1Ex := r1.GetMatchType().(*networking.StringMatch_Exact)
-				_, r2Ex := r2.GetMatchType().(*networking.StringMatch_Exact)
-				// TODO: default at the end
-				if r1Ex && !r2Ex {
-					return true
-				}
-				return false
-			})
 		} else {
 			ingressByHost[host] = &virtualServiceConfig
 		}
+
```

Comment:
for backwards compat we will need a flag for old behavior. Can combine it with https://github.com/istio/istio/pull/36855#pullrequestreview-855690740 maybe

Reply:
Done